lua/ttt2/libraries/events.lua

```diff
@@ -358,32 +279,38 @@ else --CLIENT
 	end)
 end
 
--- load the events itself
-fileloader.LoadFolder(
+local function ShouldInherit(t, base)
+	return t.base ~= t.type
+end
+
+eventTypes = classbuilder.BuildFromFolder(
 	"terrortown/gamemode/shared/events/",
-	false, -- deepsearch
 	SHARED_FILE,
-	function(path)
-		events.Initialize(path)
+	"EVENT", -- class scope
+	function(class, path, name) -- on class loaded
+		class.type = name
+		class.base = class.base or "base_event"
+
+		_G["EVENT_" .. string.upper(name)] = name
 
 		MsgN("Added TTT2 event file: ", path)
 	end,
```

Comment:
That function is repeating as well, maybe add an option to accept a table of multiple paths into `classbuilder.BuildFromFolder`?

Reply:
I thought about removing (in a future PR) all duplicated paths. Right now we have always a `shared/xyz` and a `lua/terrortown/xyz` folder. I'd prefer to only keep the latter.

Reply:
Then probably localizing one of these two exact function calls while accepting the path as an arg and calling this function manually two times with the two paths